projects/packages/connection/src/class-package-version-tracker.php

```diff
@@ -108,4 +123,20 @@ protected function update_package_versions_option( $package_versions ) {
 			set_transient( self::CACHED_FAILED_REQUEST_KEY, time(), self::CACHED_FAILED_REQUEST_EXPIRATION );
 		}
 	}
+
+	/**
+	 * Check if version check is being rate limited, and update the rate limiting transient if needed.
+	 *
+	 * @return bool
+	 */
+	private function is_rate_limiting() {
+		$last_version_check = get_transient( static::RATE_LIMITER_KEY );
+		if ( $last_version_check && time() < $last_version_check + static::RATE_LIMITER_TIMEOUT ) {
```

Comment:
Why do we need to check the transient value? I wonder if we could simplify this by just checking if the transient exists since it has an expiration anyways 🤔

Reply:
That's a good point.
I mean it as an additional check in case transients let us down, but looking at the `get_transient()` source code it does look pretty reliable, so I removed that additional expiration check.